keps/808-GangConfig/README.md

```diff
@@ -0,0 +1,444 @@
+
+The Job controller should not create a new Workload or change the workloadRef in a pod spec if the JobSet controller has already set it. To differentiate the levels of gang scheduling (JobSet, ReplicatedJob or Job replica gangs), the JobSet controller will set different PodGroup and PodGroupReplicaIndex fields in the WorkloadReference spec per pod i.e.:
+At the JobSet level, each pod spec is identical i.e. all pods have a workloadRef object reference with the same pod group name and PodGroupReplicaIndex is unset.
+At the ReplicatedJob level, all pod specs across all Job replicas for that ReplicatedJob are identical i.e. pods have a workloadRef object reference with a different pod group name for pods across different ReplicatedJobs and PodGroupReplicaIndex is unset.
+At the replica/Job level, pod specs for a single Job replica are identical i.e. pods have a workloadRef object reference with a different pod group name for pods across different ReplicatedJobs and PodGroupReplicaIndex is set to the Job replica index.
+
+See [Associating Pods into PodGroups](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4671-gang-scheduling#associating-pod-into-podgroups) for the `WorkloadReference` definition.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user, I want to schedule all workers in my workload to start at once i.e. all pods in my JobSet should be scheduled at once or not at all.
+
+The API below would fit my purpose:
+
+```
+apiVersion: jobset.x-k8s.io/v1
+kind: JobSet
+metadata:
+  name: sample-jobset
+spec:
+  scheduleAtOnce: true
+  replicatedJobs:
+    - name: replicated-job-1
+      replicas: 2
+      template:
+        spec:
+          parallelism: 4
+          completions: 4
+          template:
+            spec:
+              containers:
+                - name: initializer
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'initializer runs for 10 seconds' && sleep 10"
+    - name: replicated-job-2
+      replicas: 2
+      template:
+      spec:
+          parallelism: 4
+          completions: 4
+          template:
+            spec:
+              containers:
+                - name: launcher
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'launcher runs for 20 seconds' && sleep 20"
+                  startupProbe:
+                    exec:
+                      command:
+                        - echo
+                        - "started"
+                    initialDelaySeconds: 5
+```
```

Comment:
Will this example results in all pods pointing to a Workload defined as the following? 

```yaml
apiVersion: scheduling.k8s.io/v1alpha1
kind: Workload
metadata:
  name: sample-jobset
spec:
  podGroups:
  - name: podgroup
    policy:
      kind: Gang
      gang:
        minCount: 16
```